Client/Frontend/Browser/Favorites/FavoritesViewController.swift

```diff
@@ -34,6 +34,35 @@ private class FavoritesHeaderView: UICollectionReusableView {
 
 class FavoritesViewController: UIViewController {
     
+    /// CoreData's diffable method relies on managed object ID only.
+    /// This does not work in our case since the object may be the same but with changed title or order.
+    /// This struct stores managed object ID as well as properties that we observer whether they have changed.
+    /// Note: `order` property does not have to be stored, favorite's frc handles order updates, it returns items in correct order.
+    private struct FavoriteDiffable: Hashable {
+        let objectID: NSManagedObjectID
+        let title: String?
+        let url: String?
+    }
```

Comment:
Does the title/url affect the identity of the favourite or is identity specific to the object ID?

If it doesn't affect identity, you should write a `hash(into:)` method that only hashes the `objectID`

Reply:
If you change title or url, the objectID stays the same, it's the same object, that's the problem

I was afraid to add any hashable/equatable stuff directly into `Favorite` class to not break it in any other place

Reply:
Right then in that case its identity is only linked to `objectID` and thus the hash produced should only be based on that. This is important for diffable data sources since it uses hash to determine identity (what cells get moved), and equality to determine if that cell needs to be updated

Reply:
ah I see, so objectID must be hashed only but for equality i put the other properties

I will add it

Reply:
Yeah Hashable conforms to Equatable, you only need to implement an actual `hash(into:)` method, whereas Equality can be synthesized by the compiler with no change needed